### ISSUE TITLE
When training on the TPU, loss  is in PerReplica dictionary format. Any way to convert it in normal numpy/tensor format?

### ISSUE THREAD

**Issue by coreqode** [tensorflow/tensorflow#44973]
![image](https://user-images.githubusercontent.com/18486587/99510460-1dbdef80-29ad-11eb-9b81-ce211054c542.png)

I am returning the loss, from train_step function like this:

    def train_step(self, data, model_inputs):
        with tf.GradientTape() as tape:
            predictions = self.model(model_inputs['inputs'], training=True)
            losses = self.loss_func(data,  predictions)
            loss = sum(losses.values()) / self.train_batch_size
        
        gradients = tape.gradient(loss, self.model.trainable_variables)
        self.optimizer.apply_gradients(
            list(zip(gradients, self.model.trainable_variables))
        )

        self.update_metrics_meter(data, predictions)
        self.update_loss_meter(losses, self.train_batch_size)
        return losses

**Comment by amahendrakar** [comment]
@coreqode,
In order to reproduce the issue reported here, could you please provide the TensorFlow version, the complete code and the dataset you are using. Thanks!

**Comment by coreqode** [comment]
Thanks for quick reply.
I am using, colab tpu with tensorflow version 2.3.0. The snippet of the training loop and other functions can be found [here](https://github.com/coreqode/tboard_issue/blob/master/base_module.py)

 I am sorry, dataset is private so can't provide that. Basically, the task is just semantic segmentation of images.

**Comment by amahendrakar** [comment]
@coreqode,
The code snippet you have linked has only the function definitions. But you're not calling the functions anywhere in the code. Could you please provide a minimal code snippet using a dummy dataset so that we can look into this.

Also, if this is not a TensorFlow bug or feature request, it is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a larger community that reads questions there. Thanks!
